streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -1006,14 +1014,22 @@ private void addTasksToStateUpdater() {
     }
 
     private void addTaskToStateUpdater(final Task task) {
+        final long nowMs = System.currentTimeMillis();
```

Comment:
Here, you need to use
```suggestion
        final long nowMs = time.milliseconds();
```
We inject the time object at creation, so that we can control time for example in tests.